clang/runtime/dpct-rt/include/dpct/lib_common_utils.hpp

```diff
@@ -168,6 +168,59 @@ inline constexpr std::size_t library_data_size[] = {
     8,                                    // real_int8_32
     8                                     // real_uint8_4
 };
+
+template <class func_t, typename... args_t>
+std::invoke_result_t<func_t, args_t...>
+catch_batch_error(int *has_execption, std::string api_names,
```

Comment:
adding an overloading like `template<class RetTy, class ...ArgsT>RetTy foo(RetTy(*)(ArgsT...) func);` may reduce the complexity when called.